docs/porting/libraries.md

```diff
@@ -1,5 +1,311 @@
+By [Phillip Carter](https://github.com/cartermp)
+
+Libraries are some of the most natural projects to port from .NET Framework to .NET Core.  This article covers all of the steps you'll need to take to get to the point where you can focus on your code.  It covers the following:
+
+- Learning about important discontinued technologies for .NET Core
+- Picking the correct .NET Platform Standard for your library
+- Understanding the basics of the .NET Core project model
+- Understanding how multitargeting works in .NET Core
+- If applicable, retargeting your code to .NET Framework 4.6.1
+- Determining the portability of your code
+- Picking the correct project system in Visual Studio based on your needs
+- Porting your tests
+- Recommended approach for porting your code
+
+It's quite a lot of content, but porting is a task that will take some time, especially if you have a large codebase.  You should also be prepared to adapt the guidance here as needed to best fit your code.  Every codebase is different, so this article attempts to frame things in a flexible way, but you may find yourself needing to diverge from the prescribed guidance.
+
+## Prerequisites
+
+This article assumes you are using Visual Studio 2015 on Windows.
+
+This article also assumes that you have understood the [overview of the porting process](overview.md) and that you have resolved any issues with [3rd party dependencies](third-party-deps.md).  If you haven't done this already, it's highly recommended that you do that before moving forward.
+
+You may also wish to use [Reverse Package Search](https://packagesearch.azurewebsites.net), a service that will find packages where a given type is defined in.
+
+## Technologies Discontinued for .NET Core
+
+There are some technologies available for .NET Framework that you may use, but are outright unavailable for .NET Core.  These are technologies that we no longer promote for .NET Framework applications, and thus did not bring to .NET Core.
+
+### App Domains
+**Why was it discontinued?** AppDomains require runtime support and are generally quite expensive. While still implemented by CoreCLR, it’s not available in .NET Native and we don’t plan on adding this capability there.
+
+**What should you use instead?** AppDomains were used for different purposes. For code isolation, we recommend processes and/or containers. For dynamic loading of assemblies, we recommend the new [AssemblyLoadContext](http://dotnet.github.io/api/System.Runtime.Loader.AssemblyLoadContext.html) class.
+
+### Remoting
+**Why was it discontinued?** The idea of .NET remoting — which is transparent remote procedure calls — has been identified as a problematic architecture. Outside of that realm, it’s also used for cross AppDomain communication. On top of that, remoting requires runtime support and is quite heavyweight.
```

Comment:
Another usage for remoting has been dynamic proxy generation. For that, we should recommend... Castle Core, I suppose, at least for now. There's a branch that's compatible with Core.